### PR TITLE
Fix my sql parser

### PR DESCRIPTION
**Pull request by HGinnerup**

### CODE REVIEW

File: Jantimizer/QueryPlanParser/Models/AnalysisResult.cs [hunk 1]
```diff
@@ -15,31 +15,23 @@ public class AnalysisResult
         public ulong ActualCardinality { get; }
         public TimeSpan ActualTime { get; }
         [Ignore]
-        public List<AnalysisResult> SubQueries { get; }
+        public AnalysisResultQueryTree QueryTree { get; }
 
-        public AnalysisResult(string name, decimal estimatedCost, ulong estimatedCardinality, ulong actualCardinality, TimeSpan actualTime)
+        public AnalysisResult(AnalysisResultQueryTree subTree)
```
Comment (kris701) [inline Jantimizer/QueryPlanParser/Models/AnalysisResult.cs:20]: Why does the only constructor for this require another model? Why cant it just be the direct property values?

File: Jantimizer/Tools/Regex/RegexHelperFunctions.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Tools.Regex
+{
+    public static class RegexHelperFunctions
+    {
+
+        public static T GetRegexVal<T>(Match regexMatch, string name)
```
Comment (kris701) [inline Jantimizer/Tools/Regex/RegexHelperFunctions.cs:14]: This is a rather weird function. Would benefit from being split into two, one being `GetRegexVal` and one `GetRegexValOrNull` instead

File: Jantimizer/QueryPlanParser/Models/AnalysisResult.cs (hunk 1 above)
Comment (kris701) [inline Jantimizer/QueryPlanParser/Models/AnalysisResult.cs:20]: Why require this being another model? Why not just the property values?

**Review by kris701 — CHANGES_REQUESTED** [review]

File: Jantimizer/Tools/Regex/RegexHelperFunctions.cs (hunk 2 above)
Comment (kris701) [reply Jantimizer/Tools/Regex/RegexHelperFunctions.cs:14]: As a sidenote, it would maybe be a good idea to restrict `T` to be `where T : IConvertible`

Comment (kris701) [reply Jantimizer/Tools/Regex/RegexHelperFunctions.cs:14]: This file would maybe also benefit from being in the `Helpers` folder instead

File: Jantimizer/QueryPlanParser/Models/AnalysisResult.cs [hunk 3]
```diff
@@ -15,31 +15,23 @@ public class AnalysisResult
         public ulong ActualCardinality { get; }
         public TimeSpan ActualTime { get; }
         [Ignore]
-        public List<AnalysisResult> SubQueries { get; }
```
[the remark this answers is not in the document]
Comment (HGinnerup) [reply Jantimizer/QueryPlanParser/Models/AnalysisResult.cs:18]: The rest of the values are uncertain to exist

**Review by kris701 — APPROVED** [review]

**Pull request merged**